dcicutils/lang_utils.py

```diff
@@ -0,0 +1,156 @@
+from .misc_utils import ignored
+import re
+
+
+class EnglishUtils:
+    """
+    In most cases you can get away without using this class, but it's here in case customization is needed.
+    The usual interfaces are available as separate functions.
+    """
+
+    SECOND = 1
+    MINUTE = 60 * SECOND
+    HOUR = 60 * MINUTE
+    DAY = 24 * HOUR
+    WEEK = 7 * DAY
+
+    _TIME_UNITS = ((WEEK, "week"), (DAY, "day"), (HOUR, "hour"), (MINUTE, "minute"))
+
+    _SPECIAL_PLURALS = {
+        "radius": "radii",
+        "spectrum": "spectra",
+        "ovum": "ova",
+        "deer": "deer",
+        "fish": "fish",
```

Comment:
I ported this from tools I wrote for another system that had such examples, but they offer illustration of how English is irregular. I thought about leaving them out but then thought there might be biological samples from such a thing, so it was pointless to remove them.

Reply:
It's worth noting that I've had these tools since before there were online resources like wordnet that solve this in a more total way. However, those tools are big and not necessarily something you'd want to load onto a lambda just to solve a simple problem. These are fairly modest in size for a small problem that is not very common. I do think there could be ways we might use them in the app itself, and if we did that we might want to move to a fuller data set. (There are also complications when you go multi-language, because the rules of what you have to know to pick the right word are different, and the algorithms have to be different, even though sometimes the same or similar abstractions are possible.) For now I just wanted to change the mindset about thinking of every string operation as having to work as text modification.